### PR TITLE
Test/arm ctrl

### PR DESCRIPTION
**Pull request by akrzywosz**

### CODE REVIEW

File: baxter_collaboration_lib/CMakeLists.txt [hunk 1]
```diff
@@ -156,6 +156,11 @@ if(CATKIN_ENABLE_TESTING)
   catkin_add_gtest(test_ros_thread_obj test/test_ros_thread_obj.cpp)
   target_link_libraries(test_ros_thread_obj robot_utils)
 
+  ## Robot Arm Test
+  add_rostest_gtest(test_robot_arm test/test_robot_arm.test
+  test/test_robot_arm.cpp)
+  target_link_libraries(test_robot_arm robot_interface ${catkin_LIBRARIES})
```
Comment (alecive) [inline baxter_collaboration_lib/CMakeLists.txt]: do you need to link with `catkin_LIBRARIES` ? Did you try without?

File: baxter_collaboration_lib/test/test_robot_arm.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
+#include <gtest/gtest.h>
+#include "robot_interface/arm_ctrl.h"
+
+using namespace std;
+
+
+    //Potentially expand on the testcases; check to see that all edge values have been covered
```
Comment (alecive) [inline baxter_collaboration_lib/test/test_robot_arm.cpp]: Remove this comment (and reduce empty lines above it)

File: baxter_collaboration_lib/test/test_robot_arm.cpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    EXPECT_TRUE(ac.useForces()); 
+    EXPECT_TRUE(ac.useTracIK());
+    EXPECT_FALSE(ac.useCartCtrl()); 
+}
+
+TEST(ArmControlTest, testConstructorCustomValues) 
+{
+    std::string _name = "robot"; 
+    std::string _limb = "left";
+    //overrides default constructor values for these variables; these values are the opposite of the default values bool _use_forces = false;
```
Comment (alecive) [inline baxter_collaboration_lib/test/test_robot_arm.cpp]: split this in two lines. In general, try to avoid having long lines. The guideline says to use a max of 80 columns, but it is not mandatory to be always below 80 columns.

File: baxter_collaboration_lib/test/test_robot_arm.cpp [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    //overrides default constructor values for these variables; these values are the opposite of the default values bool _use_forces = false; 
+    bool _use_forces = false;
+    bool _use_robot = false;      
+    bool _use_trac_ik = false;
+    bool _use_cart_ctrl = true; 
+
+    ArmCtrl ac (_name, _limb, _use_robot, _use_forces, _use_trac_ik, _use_cart_ctrl);
+
+    EXPECT_EQ("robot", ac.getName());
+	EXPECT_EQ("left", ac.getLimb());
```
Comment (alecive) [inline baxter_collaboration_lib/test/test_robot_arm.cpp]: indentation

File: baxter_collaboration_lib/test/test_robot_arm.cpp [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+
+    //Potentially expand on the testcases; check to see that all edge values have been covered
+
+TEST(ArmControlTest, testConstructorDefaultValues)
+{
+    std::string _name = "robot"; 
+    std::string _limb = "left";
+    bool _use_forces = true; 
+    //Since we are actually not using the robot, _use_robot needs to be set to 'false' even though by default it should be true
```
Comment (alecive) [inline baxter_collaboration_lib/test/test_robot_arm.cpp]: split this in two lines. In general, try to avoid having long lines. The guideline says to use a max of 80 columns, but it is not mandatory to be always below 80 columns.

File: baxter_collaboration_lib/test/test_robot_arm.cpp [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    bool _use_forces = true; 
+    bool _use_robot = false;      
+    bool _use_trac_ik = true;
+    bool _use_cart_ctrl = false;
+
+    ArmCtrl ac (_name, _limb, _use_robot, _use_forces, _use_trac_ik, _use_cart_ctrl);
+
+    //Setter Methods
+    std::string _state = "teststate";
+    int _obj = 5; //some random obj id number
```
Comment (alecive) [inline baxter_collaboration_lib/test/test_robot_arm.cpp]: indentation of the comment (alignet with the one below)

File: baxter_collaboration_lib/test/test_robot_arm.test [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<launch>
+    <!-- Let's load the baxter URDF from the parameter server -->
+    <include file="$(find baxter_collaboration_lib)/launch/baxter_urdf.launch" />
+
+    <test test-name="test_robot_arm" pkg="baxter_collaboration_lib" type="test_robot_arm" />
+</launch>
```
Comment (alecive) [inline baxter_collaboration_lib/test/test_robot_arm.test]: I would change the filenames (and the executable) from `test_robot_arm` to `test_arm_ctrl`

**Review by alecive — CHANGES_REQUESTED** [review]

**Review by alecive — APPROVED** [review]

**Comment by alecive** [comment]
:+1: :sunglasses:
